testing/internal/e2e/tests/database/migration_test.go

```diff
@@ -0,0 +1,226 @@
+package database_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/boundary/api/credentiallibraries"
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/hashicorp/boundary/testing/internal/e2e/docker"
+	"github.com/hashicorp/boundary/testing/internal/e2e/vault"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+)
+
+func TestDatabaseMigration(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+	c, err := loadConfig()
+	require.NoError(t, err)
+
+	ctx := context.Background()
+
+	// Check local network configuration (needed for client to resolve worker)
+	output := e2e.RunCommand(ctx, "grep", e2e.WithArgs("-i", "127.0.0.1       localhost       boundary", "/etc/hosts"))
+	require.NoError(t, output.Err, "/etc/hosts does not contain the following entry...\n127.0.0.1       localhost       boundary")
+
+	// Create Docker network so that containers can communicate with each other
+	networkName := "e2e-network"
```

Comment:
You might want to use `t.Name()` in here, if this pattern is followed for other tests, and they run in parallel, you will want unique network names per test.